client/src/extension.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  TransportKind,
+} from 'vscode-languageclient'
+import { ExtensionContext, workspace } from 'vscode'
+import * as path from 'path'
+
+let client: LanguageClient
```

Comment:
To have it save, this should normally be `LanguageClient | undefined`, because it's not yet initialized. You might want to turn on `strict: true` in the tsconfig to have strict mode, which will let these things bubble up.